pkg/api/object.go

```diff
@@ -89,13 +83,155 @@ func (h *Handler) GetObjectStoreConnectionInfo(w http.ResponseWriter, r *http.Re
 	s3Info := model.ObjectStoreS3Info{
 		Host:       host,
 		IPEndpoint: ipEndpoint,
-		AccessKey:  accessKey,
-		SecretKey:  secretKey,
 	}
 
 	FormatJsonResponse(w, s3Info)
 }
 
+// ListUsers lists the users of the object store in this cluster.
+// GET
+// /objectstore/users
+func (h *Handler) ListUsers(w http.ResponseWriter, r *http.Request) {
+	userNames, err := rgw.ListUsers(h.context, h.AdminCommandRunner)
+	if err != nil {
+		logger.Errorf("Error listing users: %+v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	users := []model.ObjectUser{}
+	for _, userName := range userNames {
+		user, err := rgw.GetUser(h.context, h.AdminCommandRunner, userName)
+		if err != nil {
+			logger.Errorf("Error listing users: %+v", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		users = append(users, user)
+	}
+
+	FormatJsonResponse(w, users)
+}
+
+// GetUser gets the passed users info from the object store in this cluster.
+// GET
+// /objectstore/users/{USER_ID}
+func (h *Handler) GetUser(w http.ResponseWriter, r *http.Request) {
+	id := mux.Vars(r)["id"]
+
+	user, err := rgw.GetUser(h.context, h.AdminCommandRunner, id)
+	if err != nil {
+		logger.Errorf("Error gettin user (%s): %+v", id, err)
+
+		if err.Error() == "user not found" {
```

Comment:
rather than checking the error string, can you check for the type of error or have the GetUser method return a flag indicating as much?